registry/remote/auth/client.go

```diff
@@ -378,6 +378,18 @@ func (c *Client) fetchOAuth2Token(ctx context.Context, realm, service string, sc
 	return "", fmt.Errorf("%s %q: empty token returned", resp.Request.Method, resp.Request.URL)
 }
 
+var ErrInvalidCredential error = errors.New("invalid credential")
+
+// StaticCredential specifies static credentials for the given registry.
+func StaticCredential(host string, cred Credential) func(ctx context.Context, target string) (Credential, error) {
+	return func(_ context.Context, target string) (Credential, error) {
+		if target == host {
+			return cred, nil
+		}
+		return EmptyCredential, ErrInvalidCredential
```

Comment:
nit: Do we want to include the host name in the error message?

Reply:
Added the target name to the error message.